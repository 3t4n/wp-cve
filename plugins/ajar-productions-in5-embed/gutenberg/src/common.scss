/**
 * #.# Common SCSS
 *
 * Can include things like variables and mixins
 * that are used across the project.
*/

// Colors.
$black: rgb(41, 41, 41);
$white: #f4f4f4;
$gray: #dedede;
$green: #bada55;
$red: orangered;
$dark-gray-800: #23282d;
$light-gray-500: #e2e4e7; // Good for "grayed" items and borders.

// Fonts
$editor-html-font: Menlo, Consolas, monaco, monospace;
$text-editor-font-size: 14px;
