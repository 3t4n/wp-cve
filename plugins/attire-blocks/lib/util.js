import slugify from 'slugify';
import striptags from 'striptags';
import * as backgrounds from "../blocks/09-icon/backgrounds";

const {__} = wp.i18n;

const {
    getBlocks,
} = wp.data.select('core/block-editor');

class Util {
    constructor() {

    }

    static hexToRgba(hex, alpha) {
        if (!hex || hex === 'transparent') return 'transparent';
        var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
        const rgba = result ? {
            r: parseInt(result[1], 16), g: parseInt(result[2], 16), b: parseInt(result[3], 16), a: alpha / 100
        } : null;
        return "rgba( " + rgba.r + ", " + rgba.g + ", " + rgba.b + ", " + rgba.a + ")";
    }

    static defaultColor() {

        return [{color: '#ffffff', name: 'white'}, {color: '#000000', name: 'black'}, {
            color: '#3373dc',
            name: 'royal blue'
        }, {color: '#209cef', name: 'sky blue'}, {color: '#2BAD59', name: 'green'}, {
            color: '#ff3860',
            name: 'pink'
        }, {color: '#7941b6', name: 'purple'}, {color: '#F7812B', name: 'orange'}, {
            color: 'transparent',
            name: 'Transparent'
        },];
    }

    static titleCase(s) {
        return s.replace(/^_*(.)|_+(.)/g, (s, c, d) => c ? c.toUpperCase() : ' ' + d.toUpperCase())
    }

    static guidGenerator() {
        return (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1) + (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
    }

    static getBgOverlay(attributes, prefix = '') {
        if (!attributes[prefix + 'ColorLeft'] || !attributes[prefix + 'ColorRight']) {
            return 'none';
        }
        const rgba1 = Util.hexToRgba(attributes[prefix + 'ColorLeft'], attributes[prefix + 'Alpha']);
        const rgba2 = Util.hexToRgba(attributes[prefix + 'ColorRight'], attributes[prefix + 'Alpha']);
        const cs1 = attributes[prefix + 'CS1'] + "%";
        const cs2 = attributes[prefix + 'CS2'] + "%";
        return "linear-gradient( " + attributes[prefix + 'GradAngle'] + "deg, " + rgba1 + " " + cs1 + ", " + rgba2 + " " + cs2 + ")";
    }

    static getSpacingStyles(props, prefix = '') {
        const {attributes} = props;
        let style = {};
        style ['padding'] = `${attributes[prefix + 'Padding'][0]}px ${attributes[prefix + 'Padding'][1]}px ${attributes[prefix + 'Padding'][2]}px ${attributes[prefix + 'Padding'][3]}px`;
        style ['margin'] = `${attributes[prefix + 'Margin'][0]}px ${attributes[prefix + 'Margin'][1]}px ${attributes[prefix + 'Margin'][2]}px ${attributes[prefix + 'Margin'][3]}px`;
        return style;
    }

    static getSpacingV2Styles(attributes, prefix = '') {
        let style = {};
        let padding = prefix + 'Padding';
        let margin = prefix + 'Margin';
        style ['padding'] = `${this.getValueByIndex(attributes, padding, 0)} ${this.getValueByIndex(attributes, padding, 1)} ${this.getValueByIndex(attributes, padding, 2)} ${this.getValueByIndex(attributes, padding, 3)}`;
        style ['margin'] = `${this.getValueByIndex(attributes, margin, 0)} ${this.getValueByIndex(attributes, margin, 1)} ${this.getValueByIndex(attributes, margin, 2)} ${this.getValueByIndex(attributes, margin, 3)}`;
        return style;
    }

    static getSpacingV2StylesFlipped(attributes, prefix = '') {
        let style = {};
        let padding = prefix + 'Padding';
        let margin = prefix + 'Margin';
        style ['padding'] = `${this.getValueByIndex(attributes, padding, 3)} ${this.getValueByIndex(attributes, padding, 2)} ${this.getValueByIndex(attributes, padding, 1)} ${this.getValueByIndex(attributes, padding, 0)}`;
        style ['margin'] = `${this.getValueByIndex(attributes, margin, 3)} ${this.getValueByIndex(attributes, margin, 2)} ${this.getValueByIndex(attributes, margin, 1)} ${this.getValueByIndex(attributes, margin, 0)}`;
        return style;
    }

    static getValueByIndex(attributes, propName, index) {
        if (attributes[propName + 'Unit'] && (attributes[propName + 'Unit'][index] === 'auto')) {
            return 'auto';
        }
        if (!attributes[propName][index] || !attributes[propName + 'Unit']) return '0px';
        return `${attributes[propName][index]}${attributes[propName + 'Unit'][index]}`;
    }

    static getBorderStylesInline(attributes) {
        let style = {};
        style ['borderColor'] = `${(attributes.BorderColor || 'grey')}`;
        style ['borderRadius'] = `${(attributes.BorderRadius || 0)}px`;
        style ['borderStyle'] = `${(attributes.BorderStyle || 'solid')}`;
        style ['borderWidth'] = `${(attributes.BorderWidth || 0)}px`;
        return style;
    }

    /**
     * Get all block IDs.
     *
     * @param {Array} excludeId exclude block client id.
     * @param {Array} blocks blocks list to check.
     *
     * @return {Array} block anchors and slugs array.
     */
    static getAllSlugs(excludeId, blocks = 'none') {
        let slugs = [];

        if ('none' === blocks) {
            blocks = getBlocks();
        }

        blocks.forEach((block) => {
            if (block.clientId !== excludeId && block.attributes) {
                if (block.attributes.anchor) {
                    slugs.push(block.attributes.anchor);
                }
                if ('attire-blocks/tab' === block.name && block.attributes.slug) {
                    slugs.push(block.attributes.slug);
                }
            }

            if (block.innerBlocks && block.innerBlocks.length) {
                slugs = [...slugs, ...this.getAllSlugs(excludeId, block.innerBlocks),];
            }
        });

        return slugs;
    }

    /**
     * Check if slug is unique.
     *
     * @param {String} slug new slug.
     * @param {Array} slugs slugs list to check.
     *
     * @return {Boolean} is unique.
     */
    static isUniqueSlug(slug, slugs) {
        let isUnique = true;

        slugs.forEach((thisSlug) => {
            if (thisSlug === slug) {
                isUnique = false;
            }
        });

        return isUnique;
    }

    /**
     * Get slug from title.
     *
     * @param {String} title title string.
     *
     * @return {String} slug.
     */
    static getSlug(title) {
        return slugify(striptags(title), {
            replacement: '-', remove: /[*_+~()'"!?/\-—–−:@^|&#.,;%<>{}]/g, lower: true,
        });
    }

    /**
     * Get unique slug from title.
     *
     * @param {String} title title string.
     * @param {String} excludeBlockId exclude block id to not check.
     *
     * @return {String} slug.
     */
    static getUniqueSlug(title, excludeBlockId) {
        let newSlug = '';
        let i = 0;
        const allSlugs = this.getAllSlugs(excludeBlockId);

        while (!newSlug || !this.isUniqueSlug(newSlug, allSlugs)) {
            if (newSlug) {
                i += 1;
            }
            newSlug = `${this.getSlug(title)}${i ? `-${i}` : ''}`;
        }
        return newSlug;
    }

    static serialize(obj) {
        let str = [];
        for (let p in obj) if (obj.hasOwnProperty(p)) {
            str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p]));
        }
        return str.join("&");
    }

    static getValuesFromMultiSelectObject(array) {
        let data = [];
        array.forEach(item => {
            data.push(item.value);
        });
        return data;
    }

    static getRandomIcons(arr, n = 100) {
        let result = new Array(n), len = arr.length, taken = new Array(len);
        if (n > len) throw new RangeError("getRandom: more elements taken than available");
        while (n--) {
            let x = Math.floor(Math.random() * len);
            result[n] = arr[x in taken ? taken[x] : x];
            taken[x] = --len in taken ? taken[len] : len;
        }
        return result;
    }

    //Used in Icons and Social Share block
    static generateIconCss(attributes, id, class_name = 'atbs-icon') {
        let style = `
       .${class_name}-${id} span{
           background-color:${(attributes.backgroundColor || 'transparent')}!important;
           border:${attributes.BorderWidth}px ${attributes.BorderStyle} ${(attributes.BorderColor || 'transparent')};
           border-radius:${attributes.BorderRadius}px!important;
           padding:${attributes.Padding[0]}px ${attributes.Padding[1]}px ${attributes.Padding[2]}px ${attributes.Padding[3]}px !important;
           margin:${attributes.Margin[0]}px ${attributes.Margin[1]}px ${attributes.Margin[2]}px ${attributes.Margin[3]}px !important;}
       .${class_name}-${id} i{color:${(attributes.iconColor || '#000')}!important;
           font-size:${attributes.iconSize} !important;}`;

        if (attributes.iconHoverColor) {
            style += `.${class_name}-${id} span:hover i{color:${attributes.iconHoverColor}!important;}`;
        }
        if (attributes.backgroundImage) {
            style += `.${class_name}-${id} span{
            mask-repeat: no-repeat; 
            -webkit-mask-repeat: no-repeat;
            mask-size: contain; 
            -webkit-mask-size: contain; display: inline-block !important; 
            mask-position: center;
            -webkit-mask-position: center;
            -webkit-mask-image: url(${backgrounds[attributes.backgroundImage]});
            mask-image: url(${backgrounds[attributes.backgroundImage]});}`;
        }
        return style;
    }

    static textAlignOptions() {
        return [{
            label: __('Left', 'attire-blocks'), value: 'left',
        }, {
            label: __('Center', 'attire-blocks'), value: 'center',
        }, {
            label: __('Right', 'attire-blocks'), value: 'right',
        }];
    }


    static socialShareLinks() {
        return {
            blogger: {
                link: "https://www.blogger.com/blog_this.pyra?t&amp;u=", label: 'Blogger', icon: 'fab fa-blogger'
            },
            buffer: {link: 'https://buffer.com/add?url=', label: 'Buffer', icon: 'fab fa-buffer'},
            digg: {link: "http://digg.com/submit?url=", label: 'Digg', icon: 'fab fa-digg'},
            email: {link: "mailto:?body=", label: 'Email', icon: 'fas fa-envelope'},
            facebook: {link: "https://www.facebook.com/sharer.php?u=", label: 'Facebook', icon: 'fab fa-facebook'},
            google: {
                link: "https://plus.google.com/share?url=", label: 'Google Currents', icon: 'fab fa-google-plus-g'
            },
            linkedin: {link: "https://www.linkedin.com/shareArticle?url=", label: 'LinkedIn', icon: 'fab fa-linkedin'},
            odnoklassniki: {
                link: 'https://connect.ok.ru/offer?url=',
                label: 'Odnoklassniki (Одноклассники)',
                icon: 'fab fa-odnoklassniki'
            },
            pinterest: {
                link: "https://pinterest.com/pin/create/link/?url=", label: 'Pinterest', icon: 'fab fa-pinterest'
            },
            pocket: {link: 'https://getpocket.com/edit?url=', label: 'Pocket', icon: 'fab fa-get-pocket'},
            reddit: {link: "https://reddit.com/submit?url=", label: 'Reddit', icon: 'fab fa-reddit'},
            skype: {link: 'https://web.skype.com/share?url=', label: 'Skype', icon: 'fab fa-skype'},
            stumbleupon: {
                link: "https://www.stumbleupon.com/submit?url=", label: 'StumbleUpon', icon: 'fab fa-stumbleupon'
            },
            telegram: {link: 'https://telegram.me/share/url?url=', label: 'Telegram', icon: 'fab fa-telegram'},
            tumblr: {
                link: "https://www.tumblr.com/widgets/share/tool?canonicalUrl=", label: 'Tumblr', icon: 'fab fa-tumblr'
            },
            twitter: {link: "https://twitter.com/share?url=", label: 'Twitter', icon: 'fab fa-twitter'},
            vk: {link: 'https://vkontakte.ru/share.php?url=', label: 'VK', icon: 'fab fa-vk'},
            whatsapp: {link: 'https://api.whatsapp.com/send?text=', label: 'WhatsApp', icon: 'fab fa-whatsapp'},
            xing: {link: 'https://www.xing.com/app/user?op=share&url=', label: 'XING', icon: 'fab fa-xing'},
        }
    }

    static spacing_css(attributes, prefix = '') {
        let css = {}
        css['padding'] = attributes[prefix + 'Padding'][0] + 'px ' + attributes[prefix + 'Padding'][1] + 'px ' + attributes[prefix + 'Padding'][2] + 'px ' + attributes[prefix + 'Padding'][3] + 'px';
        css['margin'] = attributes[prefix + 'Margin'][0] + 'px ' + attributes[prefix + 'Margin'][1] + 'px ' + attributes[prefix + 'Margin'][2] + 'px ' + attributes[prefix + 'Margin'][3] + 'px';
        return css;
    }

    static position_css(attributes, prefix = '') {
        return {
            top: this.getValueByIndex(attributes, prefix + 'Position', 0),
            right: this.getValueByIndex(attributes, prefix + 'Position', 1),
            bottom: this.getValueByIndex(attributes, prefix + 'Position', 2),
            left: this.getValueByIndex(attributes, prefix + 'Position', 3),
        }
    }

    static spacing_css_flipped(attributes, prefix = '') {
        let css = {}
        css['padding'] = attributes[prefix + 'Padding'][0] + 'px ' + attributes[prefix + 'Padding'][3] + 'px ' + attributes[prefix + 'Padding'][2] + 'px ' + attributes[prefix + 'Padding'][1] + 'px';
        css['margin'] = attributes[prefix + 'Margin'][0] + 'px ' + attributes[prefix + 'Margin'][3] + 'px ' + attributes[prefix + 'Margin'][2] + 'px ' + attributes[prefix + 'Margin'][1] + 'px';
        return css;
    }

    static borderCss(attributes, prefix = '') {
        return {
            'border': attributes[prefix + 'BorderWidth'] + 'px ' + attributes[prefix + 'BorderStyle'] + ' ' + attributes[prefix + 'BorderColor'],
            'borderRadius': attributes[prefix + 'BorderRadius'] + 'px'
        }
    }

    static unSlash(site) {
        return site.replace(/\/$/, "");
    }

    static typographyCss(attributes, prefix = '', ignore = []) {
        let css = {};

        if (ignore.indexOf('FontSize') === -1) {
            css['fontSize'] = attributes[prefix + 'FontSize'] + attributes[prefix + 'FontSizeUnit'];
        }
        if (ignore.indexOf('FontWeight') === -1) {
            css['fontWeight'] = attributes[prefix + 'FontWeight'];
        }
        if (ignore.indexOf('LineHeight') === -1) {
            css['lineHeight'] = attributes[prefix + 'LineHeight'] + attributes[prefix + 'LineHeightUnit'];
        }
        if (ignore.indexOf('LetterSpacing') === -1) {
            css['letterSpacing'] = attributes[prefix + 'LetterSpacing'] + attributes[prefix + 'LetterSpacingUnit'];
        }
        if (ignore.indexOf('TextAlign') === -1) {
            css['textAlign'] = attributes[prefix + 'TextAlign'];
        }
        if (ignore.indexOf('FontStyle') === -1) {
            css['fontStyle'] = attributes[prefix + 'FontStyle'];
        }
        if (ignore.indexOf('TextColor') === -1) {
            css['color'] = attributes[prefix + 'TextColor'];
        }
        if (ignore.indexOf('TextTransform') === -1) {
            css['textTransform'] = attributes[prefix + 'TextTransform'];
        }

        return css;
    }

    static JSToCSS(JS) {
        let cssString = "";
        for (let objectKey in JS) {
            cssString += objectKey.replace(/([A-Z])/g, (g) => `-${g[0].toLowerCase()}`) + ": " + JS[objectKey] + ";\n";
        }

        return cssString;
    };

    static isValidHttpUrl(string) {
        let url;

        try {
            url = new URL(string);
        } catch (_) {
            return false;
        }

        return url.protocol === "http:" || url.protocol === "https:";
    }

}

export default Util;