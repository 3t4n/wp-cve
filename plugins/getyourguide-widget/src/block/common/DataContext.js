import { createContext } from 'react';

export default createContext({
  currency: {
    USD: {
      value: 'USD',
      label: 'US$ US Dollar',
    },
    EUR: {
      value: 'EUR',
      label: '€ Euro',
    },
    AUD: {
      value: 'AUD',
      label: 'AU$ Australian Dollar',
    },
    GBP: {
      value: 'GBP',
      label: '￡ British Pound',
    },
    CAD: {
      value: 'CAD',
      label: 'C$ Canadian Dollar',
    },
    CSK: {
      value: 'CSK',
      label: 'Kč Czech koruna',
    },
    DKK: {
      value: 'DKK',
      label: 'DKK Danish Krone',
    },
    HKD: {
      value: 'HKD',
      label: 'HK$ Hong Kong Dollar',
    },
    JPY: {
      value: 'JPY',
      label: '¥ Japanese Yen',
    },
    NZD: {
      value: 'NZD',
      label: 'NZ$ New Zealand Dollar',
    },
    NOK: {
      value: 'NOK',
      label: 'NOK Norwegian krone',
    },
    PLN: {
      value: 'PLN',
      label: 'zł Polish Złoty',
    },
    SGD: {
      value: 'SGD',
      label: 'S$ Singapore Dollar',
    },
    SEK: {
      value: 'SEK',
      label: 'SEK Swedish Krona',
    },
    CHF: {
      value: 'CHF',
      label: 'CHF Swiss Franc',
    },
    AED: {
      value: 'AED',
      label: 'د.إ UAE Dirham',
    },
  },
  locale_code: {
    'da-DK': { value: 'da-DK', label: 'Dansk' },
    'de-DE': { value: 'de-DE', label: 'Deutsch' },
    'de-AT': { value: 'de-AT', label: 'Deutsch (Österreich)' },
    'de-CH': { value: 'de-CH', label: 'Deutsch (Schweiz)' },
    'en-US': { value: 'en-US', label: 'English' },
    'en-GB': { value: 'en-GB', label: 'English (United Kingdom)' },
    'es-ES': { value: 'es-ES', label: 'Español' },
    'es-MX': { value: 'es-MX', label: 'Español (México)' },
    'fr-FR': { value: 'fr-FR', label: 'Français' },
    'it-IT': { value: 'it-IT', label: 'Italiano' },
    'nl-NL': { value: 'nl-NL', label: 'Nederlands' },
    'no-NO': { value: 'no-NO', label: 'Norsk' },
    'pl-PL': { value: 'pl-PL', label: 'Polski' },
    'pt-PT': { value: 'pt-PT', label: 'Português' },
    'pt-BR': { value: 'pt-BR', label: 'Português (Brasil)' },
    'fi-FI': { value: 'fi-FI', label: 'Suomen kieli' },
    'sv-SE': { value: 'sv-SE', label: 'Svenska' },
    'tr-TR': { value: 'tr-TR', label: 'Türkçe' },
    'ru-RU': { value: 'ru-RU', label: 'Русский' },
    'ja-JP': { value: 'ja-JP', label: '日本語' },
    'zh-CN': { value: 'zh-CN', label: '简体中文' },
    'zh-TW': { value: 'zh-TW', label: '繁體中文' },
  },
});
