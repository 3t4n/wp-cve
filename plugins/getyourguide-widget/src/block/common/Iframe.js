// @flow
import React from 'react';
import { stringify } from 'query-string';
import throttle from 'lodash.throttle';
import {
  activitiesDefaultAttributes, cityDefaultAttributes, gygData, spacers,
} from '../config';

const iframeMsgListener = (target) => {
  const el = target;
  if (!el) {
    return;
  }
  window.addEventListener(
    'message',
    (msg) => {
      if (
        msg.origin !== 'https://widget.getyourguide.com'
        || parseInt(el.style.height, 10) === msg.data.height
        || msg.data.url !== el.firstChild.src
      ) {
        return;
      }
      el.style.height = `${msg.data.height + 1}px`;
    },
    false,
  );
};

const postMessage = (target) => {
  if (!target) {
    return;
  }
  target.firstChild.contentWindow.postMessage('update height', 'https://widget.getyourguide.com');
};

const handleIframeResize = (target) => {
  window.addEventListener('resize', throttle(postMessage.bind(null, target), 1000));
};

const iframeHandlers = (target) => {
  if (!target) {
    return;
  }
  iframeMsgListener(target);
  postMessage(target);
  handleIframeResize(target);
};

const qs = ({ widgetType = 'activities', queries = {} }) => {
  const { partnerID: partner_id } = gygData; // eslint-disable-line camelcase
  const allQueries = (() => {
    switch (widgetType) {
    case 'city':
      return { ...cityDefaultAttributes, ...queries, partner_id };
    default:
      return { ...activitiesDefaultAttributes, ...queries, partner_id };
    }
  })();
  return stringify(
    Object.keys(allQueries).reduce(
      (acc, key) => (allQueries[key] && allQueries[key].length > 0 ? { [key]: allQueries[key], ...acc } : acc),
      {},
    ),
  );
};
const title = widgetType => `GetYourGuide ${widgetType} widget`;
const url = 'https://widget.getyourguide.com/default/';

type IframeProps = {
  queries?: Object,
  widgetType?: string,
  wpWidgetType?: string,
};
class Iframe extends React.Component<IframeProps> {
  container = React.createRef();

  static defaultProps = {
    queries: {},
    widgetType: 'activities',
    wpWidgetType: 'wp_activities',
  };

  componentDidMount() {
    const { current } = this.container;
    iframeHandlers(current);
  }

  render() {
    const { container } = this;
    const { widgetType = 'activities', wpWidgetType = 'wp_activities', queries } = this.props;
    return (
      <div ref={container}>
        <iframe
          src={`${url}${widgetType}.frame?widget=${wpWidgetType}&${qs({
            widgetType,
            queries,
          })}`}
          style={{ height: '100%', marginBottom: spacers.standard }}
          title={title(widgetType)}
        />
      </div>
    );
  }
}

const wpIframe = ({ widgetType, wpWidgetType, queries }) => {
  const { createElement } = window.wp.element;
  return createElement(
    'div',
    {
      class: 'gyg-iframe-container',
      style: {
        marginBottom: spacers.standard,
      },
    },
    createElement('iframe', {
      style: {
        border: '0',
        height: '100%',
        width: '100%',
      },
      title: title(widgetType),
      src: `${url}${widgetType}.frame?widget=${wpWidgetType}&${qs({
        widgetType,
        queries,
      })}`,
    }),
  );
};

export { Iframe, wpIframe, iframeHandlers };
