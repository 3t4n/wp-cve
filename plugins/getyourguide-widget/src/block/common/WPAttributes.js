// @flow
import { Component } from 'react';

type WPAttributesProps = {
  attributes: Object,
  children: React.Node,
  defaultAttributes: Object,
  setAttributes: Function,
};

export default class WordPress extends Component<WPAttributesProps> {
  constructor(props) {
    super(props);
    const { defaultAttributes } = this.props;
    this.labels = Object.keys(defaultAttributes).reduce((acc, key) => {
      switch (key) {
      case 'cmp':
        return { ...acc, cmp: "Track your widget's performance with a campaign name (optional)" };
      case 'iata':
        return { ...acc, iata: 'IATA Airport Code' };
      case 'currency':
        return { ...acc, currency: 'Currency' };
      case 'locale_code':
        return { ...acc, locale_code: 'Language' };
      case 'number_of_items':
        return { ...acc, number_of_items: 'Number of items' };
      case 'q':
        return { ...acc, q: 'Search query' };
      default:
        return acc;
      }
    }, {});
    this.state = defaultAttributes;
  }

  componentDidUpdate(_, prevState) {
    const { attributes, defaultAttributes, setAttributes } = this.props;
    setAttributes(
      Object.keys({ ...defaultAttributes, ...attributes }).reduce((acc, key) => {
        const { [key]: update } = this.state;
        return prevState[key] !== update ? { ...acc, [key]: update } : acc;
      }, {}),
    );
  }

  handleChange = (key = '', e = { preventDefault: () => null, target: { value: '' } }) => {
    e.preventDefault();
    this.setState({ [key]: e.target.value });
  };

  render() {
    const { labels, handleChange, state } = this;
    const { attributes, children, defaultAttributes } = this.props;
    return children({
      labels,
      state,
      attributes,
      defaultAttributes,
      handleChange,
    });
  }
}
