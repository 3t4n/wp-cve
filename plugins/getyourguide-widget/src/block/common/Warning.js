import React from 'react';

const Warning = () => (
  <strong className="notice notice-error" style={{ display: 'block' }}>
    {"You haven't entered your GetYourGuide partner ID."}
    {' '}
    <a href={`${window.location.origin}/wp-admin/options-general.php?page=getyourguide`}>
      Enter it here.
    </a>
  </strong>
);

export default Warning;
