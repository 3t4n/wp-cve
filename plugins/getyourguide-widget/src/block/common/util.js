// @flow
import React, { Fragment } from 'react';

export const Input = ({
  label,
  value,
  onChange,
  ...passThrough
}: {
  label: string,
  value: string,
  onChange: Function,
}) => (
  <div>
    <label>
      {label}
      <input
        {...{
          style: { display: 'block' },
          onChange,
          value,
          ...passThrough,
        }}
      />
    </label>
  </div>
);

export const Select = ({
  data = {},
  value,
  onChange,
  label,
}: {
  data: Object,
  value: string,
  onChange: Function,
  label: string,
}) => (
  <label style={{ display: 'block' }}>
    {label}
    <select style={{ display: 'block' }} value={value} onBlur={onChange} onChange={onChange}>
      {Object.keys(data).map((option) => {
        const { value: optionValue, label: optionLabel } = data[option];
        return (
          <option key={optionLabel} {...{ value: optionValue }}>
            {optionLabel}
          </option>
        );
      })}
    </select>
  </label>
);

export const Inputs = ({
  data,
  keys,
  labels,
  onChange,
  selects = [],
  values,
}: {
  data: Object,
  keys: Array<string>,
  labels: Object,
  onChange: Function,
  selects: Array<string>,
  values: Object,
}) => (
  <Fragment>
    {keys.map((key) => {
      if (selects.includes(key)) {
        return (
          <Select
            key={key}
            {...{
              data: data[key],
              label: labels[key],
              value: values[key],
              onChange: e => onChange(key, e),
            }}
          />
        );
      }
      return (
        <Input
          key={key}
          {...{
            label: labels[key],
            onChange: e => onChange(key, e),
            type: Number.isInteger(parseInt(values[key], 10)) ? 'number' : 'text',
            value: values[key],
          }}
        />
      );
    })}
  </Fragment>
);
