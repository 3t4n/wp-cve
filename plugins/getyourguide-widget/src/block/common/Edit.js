import React, { Component } from 'react';
import DataContext from './DataContext';
import { Iframe } from './Iframe';
import Warning from './Warning';
import { gygData } from '../config';

class Edit extends Component {
  state = {
    optionsDisplay: 'flex',
  };

  render() {
    const {
      children,
      defaultAttributes,
      attributes,
      labels,
      handleChange,
      state,
      widgetType,
    } = this.props;
    const { optionsDisplay: display } = this.state;
    const keys = Object.keys({ ...defaultAttributes, ...attributes });
    return (
      <DataContext.Consumer>
        {data => (
          <div>
            {gygData.partnerID.length === 0 && <Warning />}
            <button
              className="components-button"
              type="button"
              style={{ float: 'right' }}
              onClick={() => this.setState(({ optionsDisplay }) => ({
                optionsDisplay: optionsDisplay === 'flex' ? 'none' : 'flex',
              }))
              }
            >
              {`${display === 'none' ? 'Show' : 'Hide'} options`}
            </button>
            <div
              className="components-placeholder"
              style={{
                alignItems: 'initial',
                display,
                marginBottom: '10px',
                textAlign: 'left',
              }}
            >
              {children({
                data,
                keys,
                defaultAttributes,
                attributes,
                labels,
                handleChange,
                state,
              })}
            </div>
            <Iframe {...{ widgetType, queries: { ...defaultAttributes, ...attributes } }} />
          </div>
        )}
      </DataContext.Consumer>
    );
  }
}

export default Edit;
