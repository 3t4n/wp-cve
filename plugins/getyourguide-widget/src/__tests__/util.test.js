import React from 'react';
import renderer from 'react-test-renderer';
import { Input, Select } from '../block/common/util';

describe('Input', () => {
  it('renders correctly with no props', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Select', () => {
  it('renders correctly with no props', () => {
    const tree = renderer.create(<Select />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
