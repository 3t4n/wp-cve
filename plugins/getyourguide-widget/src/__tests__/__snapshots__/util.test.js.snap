// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly with no props 1`] = `
<div>
  <label>
    <input
      style={
        Object {
          "display": "block",
        }
      }
    />
  </label>
</div>
`;

exports[`Select renders correctly with no props 1`] = `
<label
  style={
    Object {
      "display": "block",
    }
  }
>
  <select
    style={
      Object {
        "display": "block",
      }
    }
  />
</label>
`;
